src/index.js

```diff
@@ -419,14 +419,20 @@ export default React.createClass({
               <span key={col.id}
                 style={{display: 'flex', alignContent: 'flex-end', flex: 1}}>
                 {!col.hideFilter ? (
-                  <input type='text'
-                    style={{
-                      flex: 1,
-                      width: 20
-                    }}
-                    value={filter ? filter.value : ''}
-                    onChange={(event) => this.filterColumn(column, event, col)}
-                  />
+                  _.normalizeComponent(col.filterRender,
+                    {
+                      onFilterChange: (value) => (this.filterColumn(column, value, col))
+                    },
+                    (
+                      <input type='text'
```

Comment:
Also, see that I'm sending in `filter`, and `onFilterChange` to the component, which gives more control.  In fact, we should probably send through all of the information as we do here: https://github.com/tannerlinsley/react-table#custom-cell-header-and-footer-rendering